lib/cocoapods/installer/xcode/target_validator.rb

```diff
@@ -148,6 +149,10 @@ def verify_swift_pods_have_module_dependencies
           raise Informative, 'The following Swift pods cannot yet be integrated '\
                              "as static libraries:\n\n#{error_messages.join("\n\n")}"
         end
+
+        def verify_no_multiple_project_names
+          # TODO
```

Comment:
need to integrate this.

Reply:
Does this verify that a single pod doesn't belong to two different projects? One possible configuration is a user wants to group all their iOS pods together into one project and their watch pods together. Do we want to allow or disallow this type of cofig?

Reply:
I think it already works but I will verify with a test. PodTarget are split across variants so I think this propagates through everything.

Reply:
> doesn't belong to two different projects?

I think you mean if the pod is split across different variants, not so much in the sense of multi-xcodeproj or `project` DSL in `Podfile` correct?

Reply:
Confirmed it does not throw because pod targets are already split. This means folks can group their projects by platform or anything else that pod variants support.